From gregkh@suse.de Mon Apr 28 23:10:43 2008
X-Envelope-From: <stable-commits-owner@vger.kernel.org>
X-Envelope-To: <vger@sl-h.de>
X-Delivery-Time: 1209417044
X-UID: 224283
Return-Path: <stable-commits-owner@vger.kernel.org>
X-RZG-CLASS-ID: mi
Received: from vger.kernel.org ([209.132.176.167])
	by mailin.webmailer.de (ahmed-mailin mi17) (RZmta 16.28)
	with ESMTP id T0233ek3SLABSR for <vger@sl-h.de>;
	Mon, 28 Apr 2008 23:10:43 +0200 (MEST)
	(envelope-from: <stable-commits-owner@vger.kernel.org>)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
	id S938075AbYD1VKm (ORCPT <rfc822;vger@sl-h.de>);
	Mon, 28 Apr 2008 17:10:42 -0400
Received: (majordomo@vger.kernel.org) by vger.kernel.org id S938072AbYD1VKm
	(ORCPT <rfc822;stable-commits-outgoing>);
	Mon, 28 Apr 2008 17:10:42 -0400
Received: from mail.suse.de ([195.135.220.2]:47023 "EHLO mx1.suse.de"
	rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
	id S938068AbYD1VKl (ORCPT <rfc822;stable-commits@vger.kernel.org>);
	Mon, 28 Apr 2008 17:10:41 -0400
Received: from Relay2.suse.de (relay-ext.suse.de [195.135.221.8])
	(using TLSv1 with cipher DHE-RSA-AES256-SHA (256/256 bits))
	(No client certificate requested)
	by mx1.suse.de (Postfix) with ESMTP id 9F05841177;
	Mon, 28 Apr 2008 23:10:40 +0200 (CEST)
Subject: patch x86-fix-32-bit-x86-msi-x-allocation-leakage.patch added to 2.6.24-stable tree
To: peter.p.waskiewicz.jr@intel.com,
 gregkh@suse.de
Cc: <stable@kernel.org>,
 <stable-commits@vger.kernel.org>
From:	<gregkh@suse.de>
Date:	Mon, 28 Apr 2008 14:10:43 -0700
In-Reply-To: <20080428185603.11595.17319.stgit@scrappy.jf.intel.com>
Message-Id: <20080428211039.BA30A1454476@imap.suse.de>
Sender:	stable-commits-owner@vger.kernel.org
Precedence: bulk
Reply-To: linux-kernel@vger.kernel.org
List-ID: <stable-commits.vger.kernel.org>
X-Mailing-List:	stable-commits@vger.kernel.org
Status: R
X-Status: NC
X-KMail-EncryptionState:  
X-KMail-SignatureState:  
X-KMail-MDN-Sent:  


This is a note to let you know that we have just queued up the patch titled

     Subject: x86: Fix 32-bit x86 MSI-X allocation leakage

to the 2.6.24-stable tree.  Its filename is

     x86-fix-32-bit-x86-msi-x-allocation-leakage.patch

A git repo of this tree can be found at 
    http://www.kernel.org/git/?p=linux/kernel/git/stable/stable-queue.git;a=summary


>From stable-bounces@linux.kernel.org Mon Apr 28 11:58:38 2008
From: PJ Waskiewicz <peter.p.waskiewicz.jr@intel.com>
Date: Mon, 28 Apr 2008 11:56:03 -0700
Subject: x86: Fix 32-bit x86 MSI-X allocation leakage
To: stable@kernel.org, linux-kernel@vger.kernel.org
Cc: netdev@vger.kernel.org
Message-ID: <20080428185603.11595.17319.stgit@scrappy.jf.intel.com>

From: PJ Waskiewicz <peter.p.waskiewicz.jr@intel.com>

commit 9d9ad4b51d2b29b5bbeb4011f5e76f7538119cf9 upstream

This bug was introduced in the 2.6.24 i386/x86_64 tree merge, where
MSI-X vector allocation will eventually fail.  The cause is the new
bit array tracking used vectors is not getting cleared properly on
IRQ destruction on the 32-bit APIC code.

This can be seen easily using the ixgbe 10 GbE driver on multi-core
systems by simply loading and unloading the driver a few times.
Depending on the number of available vectors on the host system, the
MSI-X allocation will eventually fail, and the driver will only be
able to use legacy interrupts.

Signed-off-by: Peter P Waskiewicz Jr <peter.p.waskiewicz.jr@intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 arch/x86/kernel/io_apic_32.c |    1 +
 1 file changed, 1 insertion(+)

--- a/arch/x86/kernel/io_apic_32.c
+++ b/arch/x86/kernel/io_apic_32.c
@@ -2478,6 +2478,7 @@ void destroy_irq(unsigned int irq)
 	dynamic_irq_cleanup(irq);
 
 	spin_lock_irqsave(&vector_lock, flags);
+	clear_bit(irq_vector[irq], used_vectors);
 	irq_vector[irq] = 0;
 	spin_unlock_irqrestore(&vector_lock, flags);
 }


Patches currently in stable-queue which might be from peter.p.waskiewicz.jr@intel.com are

queue-2.6.24/x86-fix-32-bit-x86-msi-x-allocation-leakage.patch
--
To unsubscribe from this list: send the line "unsubscribe stable-commits" in
the body of a message to majordomo@vger.kernel.org
More majordomo info at  http://vger.kernel.org/majordomo-info.html

